examples/permissioned-token/test/common.js

```diff
@@ -0,0 +1,379 @@
+const {
+  getProgram
+} = require('@algo-builder/algob');
+const { Runtime, AccountStore, types } = require('@algo-builder/runtime');
+
+const minBalance = 20e6; // 20 ALGOs
+
+const CLAWBACK = 'clawback.py';
+const CONTROLLER = 'controller.py';
+const PERMISSIONS = 'permissions.py';
+const CLEAR_STATE = 'clear_state_program.py';
+
+const ALICE_ADDRESS = 'EDXG4GGBEHFLNX6A7FGT3F6Z3TQGIU6WVVJNOXGYLVNTLWDOCEJJ35LWJY';
+
+let master = new AccountStore(10000e6);
+let alice, bob, elon;
+let runtime;
+let assetIndex;
+let lsig;
+let controllerAppID, permissionsAppId;
+
+let CLAWBACK_PROGRAM;
+let CONTROLLER_PROGRAM;
+let PERMISSIONS_PROGRAM;
+const CLEAR_STATE_PROGRAM = getProgram(CLEAR_STATE);
+
+const TRANSFER_ARG = 'str:transfer';
+
+// Sync Accounts
+function syncAccounts (runtime) {
+  master = runtime.getAccount(master.address);
+  alice = runtime.getAccount(alice.address);
+  bob = runtime.getAccount(bob.address);
+  elon = runtime.getAccount(elon.address);
+}
+
+/**
+ * - Setup token (ASA gold)
+ * - Setup Controller asc
+ * - Setup asset clawback + update asset clawback to escrow contract
+ * - Setup permissions smart contract
+ * NOTE: During setup - ASA.reserve, ASA.manager & current_permissions_manager is set as alice.address
+ */
+function setupEnv () {
+  // Create Accounts and Env
+  alice = new AccountStore(minBalance, { addr: ALICE_ADDRESS, sk: new Uint8Array(0) });
+  bob = new AccountStore(minBalance);
+  elon = new AccountStore(minBalance);
+  runtime = new Runtime([master, alice, bob, elon]);
+
+  // Deploy ASA
+  assetIndex = runtime.addAsset('gold', { creator: { ...alice.account, name: 'alice' } });
+  const asaDef = runtime.getAssetDef(assetIndex);
+
+  // Deploy Controller SSC
+  let sscFlags = {
+    sender: alice.account,
+    localInts: 0,
+    localBytes: 0,
+    globalInts: 2,
+    globalBytes: 0,
+    foreignAssets: [assetIndex]
+  };
+  CONTROLLER_PROGRAM = getProgram(CONTROLLER, { TOKEN_ID: assetIndex });
+  controllerAppID = runtime.addApp(
+    sscFlags, {}, CONTROLLER_PROGRAM, CLEAR_STATE_PROGRAM
+  );
```

Comment:
let's move it to a separate functions:
* deployController
* deployClawback
* ....

Reply:
done